src/app/components/inject-list/inject-list.component.ts

```diff
@@ -67,10 +73,10 @@ export class InjectListComponent implements OnDestroy, OnInit {
   @Input() isEditMode: boolean;
   @Input() catalog: Catalog = {};
   @Input() injectType: InjectType = {};
+  @Input() injectList: Injectm[] = [];
```

Comment:
It looks like we are passing in this list but then it gets updated from the subscriptions below. Is that what you intended?

Reply:
yes, that is intended.  There are three cases that use the list.  One with inject from a catalog, one with injects from an inject type, and one when the inject list is passed in directly